internal/resources/datastore/datastore_storage_config.go

```diff
@@ -155,9 +155,25 @@ func (r *Config) mutate(ctx context.Context, tenantControlPlane *kamajiv1alpha1.
 			username = coalesceFn(tenantControlPlane.Status.Storage.Setup.User)
 		}
 
+		var dataStoreSchema string
+		switch {
+		case len(tenantControlPlane.Spec.DataStoreSchema) > 0:
+			// for new TCPs, the spec field will have been provided by the user
+			// or defaulted by the defaulting webhook
+			dataStoreSchema = tenantControlPlane.Spec.DataStoreSchema
+		case len(tenantControlPlane.Status.Storage.Setup.Schema) > 0:
+			// for existing TCPs, the dataStoreSchema will be adopted from the status,
+			// as the mutating webhook only takes care of TCP creations, not updates
+			dataStoreSchema = tenantControlPlane.Status.Storage.Setup.Schema
+			tenantControlPlane.Spec.DataStoreSchema = dataStoreSchema
```

Comment:
```suggestion
		case len(tenantControlPlane.Status.Storage.Setup.Schema) > 0:
			// for existing TCPs, the dataStoreSchema will be adopted from the status,
			// as the mutating webhook only takes care of TCP creations, not updates
			dataStoreSchema = tenantControlPlane.Status.Storage.Setup.Schema
			tenantControlPlane.Spec.DataStoreSchema = dataStoreSchema
		case len(tenantControlPlane.Spec.DataStoreSchema) > 0:
			// for new TCPs, the spec field will have been provided by the user
			// or defaulted by the defaulting webhook
			dataStoreSchema = tenantControlPlane.Spec.DataStoreSchema
```

Should we prioritize `Status` given the state machine behavior of TenantControlPlane objects?